### PR TITLE
Verify labels attached to logs on rsyslog server

### PR DESCRIPTION
**Pull request by sasagarw**
### Description

This PR:
- Adds an e2e tests to verify that the labels are attached to the logs on rsyslog server for rfc5424 and rfc3164.

/cc @anpingli  
/assign @vimalk78 @jcantrill  

/cherry-pick

### CODE REVIEW

File: test/e2e/logforwarding/syslog/forward_to_syslog_test.go [hunk 1]
```diff
@@ -139,6 +143,17 @@ var _ = Describe("[ClusterLogForwarder] Forwards logs", func() {
...
 				expectedProcID := forwarder.Spec.Outputs[0].Syslog.ProcID
 				Expect(logStore.GrepLogs(grepprocid, helpers.DefaultWaitForLogsTimeout)).To(Equal(expectedProcID), "Expected: "+expectedProcID)
+				grepMsgContent := fmt.Sprintf(`grep %s %%s| grep pod_name | tail -n 1 | sed -n -e 's/^.*openshift://p'`, logGenPod)
+				str, err := logStore.GrepLogs(grepMsgContent, helpers.DefaultWaitForLogsTimeout)
+				Expect(err).To(BeNil())
+				str = strings.ReplaceAll(str, "=>", ":")
+				msg := map[string]map[string]string{}
+				err = json.Unmarshal([]byte(str), &msg)
+				Expect(err).To(BeNil())
+				expectedLabelSecure := forwarder.Spec.Pipelines[0].Labels["secure"]
```
Comment (jcantrill) [inline test/e2e/logforwarding/syslog/forward_to_syslog_test.go:153]: This is part of the setup which you already know that could be defined initially as a `const`.  Doing so earlier in the test will allow you to remove these lines making the test more clear.  Comment applies to all these labels

File: test/e2e/logforwarding/syslog/forward_to_syslog_test.go [hunk 2]
```diff
@@ -139,6 +143,17 @@ var _ = Describe("[ClusterLogForwarder] Forwards logs", func() {
 				Expect(logStore.GrepLogs(grepmsgid, helpers.DefaultWaitForLogsTimeout)).To(Equal(expectedMsgID), "Expected: "+expectedMsgID)
 				expectedProcID := forwarder.Spec.Outputs[0].Syslog.ProcID
 				Expect(logStore.GrepLogs(grepprocid, helpers.DefaultWaitForLogsTimeout)).To(Equal(expectedProcID), "Expected: "+expectedProcID)
+				grepMsgContent := fmt.Sprintf(`grep %s %%s| grep pod_name | tail -n 1 | sed -n -e 's/^.*openshift://p'`, logGenPod)
```
Comment (jcantrill) [inline test/e2e/logforwarding/syslog/forward_to_syslog_test.go:146]: please implement the needed method in `syslog.go`:
```
func (syslog *syslogReceiverLogStore) HasApplicationLogs(timeToWait time.Duration) (bool, error) {
	return false, fmt.Errorf("Not implemented")
}
```

**Review by jcantrill — COMMENTED** [review]
/approve
/hold

cc @vimalk78

File: test/e2e/logforwarding/syslog/forward_to_syslog_test.go (hunk 2 above)
Comment (sasagarw) [reply test/e2e/logforwarding/syslog/forward_to_syslog_test.go]: Why we need to implement this method? The logs generated by log-generator pod while testing is shipped to infra.log. There is no application logs generated during the process.

File: test/e2e/logforwarding/syslog/forward_to_syslog_test.go (hunk 1 above)
Comment (sasagarw) [reply test/e2e/logforwarding/syslog/forward_to_syslog_test.go:155]: Do you mean to define all the "expected" variables as constants above?

Comment (jcantrill) [reply test/e2e/logforwarding/syslog/forward_to_syslog_test.go:155]: reduce code duplication as much as we can where it makes sense

**Comment by jcantrill** [comment]
@vimalk78 @sasagarw is this still a PR we want to pursue?

**Pull request closed without merging**